prompting/tools/datasets/base.py

```diff
@@ -86,6 +86,41 @@ def next(
         return Context(**info)
 
 
+class BatchDataset(ABC):
+    """Base class for datasets."""
+
+    max_tries: int = 10
+
+    @abstractmethod
+    async def random_batch(self, name):
```

Comment:
```suggestion
    async def random(self, name):
```

Reply:
I'd prefer to keep the naming simple since the class is already a `BatchDataset`. If you agree then make the remaining changes across other related files